addons/hr_expense/models/res_config_settings.py

```diff
@@ -22,17 +21,31 @@ class ResConfigSettings(models.TransientModel):
     @api.model
     def get_values(self):
         res = super(ResConfigSettings, self).get_values()
+        expense_alias = self.env.ref('hr_expense.mail_alias_expense', raise_if_not_found=False)
         res.update(
-            expense_alias_prefix=self.env.ref('hr_expense.mail_alias_expense').alias_name,
+            expense_alias_prefix=expense_alias.alias_name if expense_alias else False,
         )
         return res
 
     def set_values(self):
         super().set_values()
-        alias = self.env.ref('hr_expense.mail_alias_expense', raise_if_not_found=False)
-        if alias and alias.alias_name != self.expense_alias_prefix:
-            alias.alias_name = self.expense_alias_prefix
-
-    @api.depends('use_mailgateway')
+        expense_alias = self.env.ref('hr_expense.mail_alias_expense', raise_if_not_found=False)
+        if not expense_alias and self.expense_alias_prefix:
+            # create data again
+            alias = self.env['mail.alias'].sudo().create({
+                'alias_contact': 'employees',
+                'alias_model_id': self.env['ir.model']._get_id('hr.expense'),
+                'alias_name': self.expense_alias_prefix,
+            })
+            self.env['ir.model.data'].sudo().create({
+                'name': 'mail_alias_expense',
+                'module': 'hr_expense',
+                'model': 'mail.alias',
+                'res_id': alias.id,
```

Comment:
Avoid deletion at update.
```suggestion
                'res_id': alias.id,
                'noupdate': True,
```